[Enhancement][RHEL/7] Migrate iptables XCCDF and OVAL to FirewallD

redhatrises opened the pull request:
- Add initial FirewallD XCCDF and OVAL content
- Update 'rht-cpp' profile to use FirewallD
- Update and remove iptables content as the FirewallD is the preferred firewall
  tool going forward and the service conflicts with iptables services
- Fixes #61, #62, and #63

iankko commented:
Thanks for the proposal, Gabe. Will have a look @ this.

iankko commented on RHEL/7/input/system/network/firewalld.xml:
```diff
@@ -103,8 +103,155 @@ separation of runtime and permanent configuration options. It has support for
+allows additional services to penetrate the default firewall rules
+and automatically adjusts the <tt>firewalld</tt> ruleset(s).</warning>
+
+<Rule id="set_firewalld_default_zone" severity="medium">
```
The particular rule ID here and in the referenced OVAL check below is ```set_firewalld_default_zone```, but in the ```RHT-CCP``` profile add-on there's ```set_firewalld_default_rule``` rule references (leading to ```notchecked``` result).

Change the referenced rule name in ```RHT-CCP``` profile to ```set_firewalld_default_zone```.

iankko commented on RHEL/7/input/auxiliary/stig_overlay.xml:
```diff
@@ -312,56 +312,56 @@
 		<title>The system must employ a local IPv4 firewall.</title>
 	</overlay>
-	<overlay owner="disastig" ruleid="set_iptables_default_rule" ownerid="RHEL-06-000120" disa="66" severity="medium">
+	<overlay owner="disastig" ruleid="set_firewalld_default_rule" ownerid="RHEL-06-000120" disa="66" severity="medium">
```
All the occurrences of ```set_firewalld_default_rule``` in this file should be replaced with ```set_firewalld_default_zone``` rule ID too.

redhatrises commented:
@iankko fixed

iankko commented:
@redhatrises 
> @iankko fixed

Thanks, Gabe. Attempt to build this looks the ```package_firewall_installed``` OVAL check is missing / there's another typo. Further details from RHEL-7 system:
```
[root@localhost 7]# make clean
rm -f output/*.xml output/*.html output/*.xhtml output/*.pdf  output/*.spec output/*.tar output/*.gz output/*.ini output/*.csv
rm -rf dist/content dist/guide
rm -rf build
[root@localhost 7]# pwd
/root/scap-security-guide/RHEL/7
[root@localhost 7]# make
..
File '/root/scap-security-guide/RHEL/7/output/ssg-rhel7-ds.xml' line 3692: Element '{http://oval.mitre.org/XMLSchema/oval-definitions-5}extend_definition': No match found for key-sequence ['oval:ssg:def:722'] of keyref '{http://oval.mitre.org/XMLSchema/oval-definitions-5}extendKeyRef'.
OpenSCAP Error: Invalid SCAP Source Datastream (1.2) content in /root/scap-security-guide/RHEL/7/output/ssg-rhel7-ds.xml. [oscap_source.c:205]
make: *** [content] Error 1
```
Searching for ```722``` in ```RHEL/7output/ssg.ini``` returns:
```
[root@localhost output]# grep '= 722' ssg.ini 
package_firewall_installed = 722
[root@localhost output]# cd ..
[root@localhost 7]# grep -rHn 'package_firewall_installed' *
build/rhel7_checks/service_firewalld_enabled.xml:22:    <extend_definition comment="firewall installed" definition_ref="package_firewall_installed" />
input/checks/oval_5.11/service_firewalld_enabled.xml:22:    <extend_definition comment="firewall installed" definition_ref="package_firewall_installed" />
output/unlinked-rhel7-oval.xml:3692:        <extend_definition comment="firewall installed" definition_ref="package_firewall_installed"/>
output/ssg.ini:627:package_firewall_installed = 722
[root@localhost 7]# grep -rHn 'package_firewalld_installed' *
```

Summary:
* Since ```firewalld``` is provided by ```firewalld``` RPM package (```firewalld``` it's what got added into ```packages_installed.csv``` file from look at above changes), the ```package_firewalld_installed``` OVAL check is missing (you seem to have forgotten to run 'make packages' in ```RHEL/7/input/checks/templates``` directory, and copy the ```output/package_firewalld_installed.xml``` check into ```RHEL/7/input/checks``` directory,
* Also in ```input/checks/oval_5.11/service_firewalld_enabled.xml``` you reference the ```package_firewall_installed``` rule (note the missing 'd' character).:
```
input/checks/oval_5.11/service_firewalld_enabled.xml:22:    <extend_definition comment="firewall installed" definition_ref="package_firewall_installed" />
```
So that typo should be fixed yet, prior this can be re-reviewed.

Thank you, Jan.

redhatrises commented:
@iankko ran `make` and didn't get errors. hmmmmm.....interesting. I updated to use `firewalld` everywhere. The `package_firewalld_installed.xml` is under `shared/oval`. Is `make` not pulling it in from `shared/oval`?

iankko commented:
@redhatrises 
> @iankko ran make and didn't get errors. hmmmmm.....interesting.

One reason which comes to mind why you might not be seeing that failure. Just OOC (out-of-curiosity) which openscap version were you building that content against? The point is the current openscap version which is in RHEL-7 (openscap-1.1.* IIRC) does ```NOT``` support OVAL-5.11.1 yet. To verify the proper work of OVAL-5.11.1 features it's necessary to install ```openscap``` for Simon's COPR repository:
&nbsp; &nbsp;  [1]  https://copr.fedoraproject.org/coprs/isimluk/OpenSCAP/

The openscap version from that repository (```openscap-1.2.5``` IIRC) already contains support for OVAL-5.11 language version && therefore is capable to verify also issues in ```systemd*``` probes).

The oscap version currently present in RHEL-7 doesn't support OVAL-5.11 version yet, and therefore SSG content has been modified to start building OVAL-5.11 (thus RHEL-7's ```service_*_enabled / service_*_disabled``` checks only in case the oscap version on the underlying system supports OVAL-5.11 already). The reason for this behaviour is not to see the errors when building ```5.11``` OVAL xml document version.

So you might not have seen those errors, because ```service_firewalld_enabled``` OVAL check was not even attempted to be build / included into produced OVAL if built with current RHEL-7's oscap version. I will add a warning message about this into ```Makefile``` to in a more clear way indicate this.

Therefore to build ```service_*_enabled``` checks for RHEL-7 we should be using oscap version from Simon's COPR (since that's what arrive in some of the future openscap versions in RHEL-7 and therefore we need to ensure the content to be working properly against that version).

Since OVAL-5.11 introduced some backward incompatible changes (e.g. ```systemd``` probes to mention some examples), it's not possible to attempt building RHEL-7's ```services``` related content using the oscap version, which is in RHEL-7 now (since in that case the ```services``` content wouldn't be included in the final OVAL [that's actually the reason why new ```oval-5.11``` subdirectory has been introduced under ```RHEL/7/input/checks``` folder], therefore there would be no error reported with that content - since that's what we actually want when building with current RHEL-7's oscap version. To report issues only in features supported in OVAL-5.10.1 language version).

> I updated to use firewalld everywhere. The package_firewalld_installed.xml is under shared/oval. Is make not pulling it in from shared/oval?

Have searched ```shared/oval``` but can't find ```package_firewalld_installed``` OVAL check. Maybe it has been present before, but isn't now?:
&nbsp; &nbsp; [2] https://github.com/OpenSCAP/scap-security-guide/tree/master/shared/oval

Thanks, Jan.

redhatrises commented:
> Have searched `shared/oval` but can't find `package_firewalld_installed` OVAL check. Maybe it has been present before, but isn't now?

@iankko Thanks. Added the `package_firewalld_installed` OVAL check as it was missing. Looks like a have some sort of issue with my env. I tested in a new env, got the error message, and added the `package_firewalld_installed` which fixed the error on build.

iankko commented:
@redhatrises 
> @iankko Thanks. Added the package_firewalld_installed OVAL check as it was missing. Looks like a have some sort of issue with my env. I tested in a new env, got the error message, and added the package_firewalld_installed which fixed the error on build.

Thanks Gabe! Verified this version works fine on RHEL-7 system. ACK && Merging.
